### PR TITLE
chore: Describe default type attribute in Buttons documentation

### PR DESCRIPTION
**Pull request by Ldoppea**
MUI documentation does not describe default button's `type`, nor explain that is it possible to set classic HTML attributes, so it is easy to forget applying the correct button type when inside of a form

This is what happened on a previous PR when migrating from old `react/deprecated/Button` to new `react/Buttons` as those components have opposite default behaviors

Related PR: cozy/cozy-settings#674

### CODE REVIEW

File: react/Buttons/Readme.md [hunk 1]
```diff
@@ -1,4 +1,10 @@
-The `ref` is forwarded to the root element
```
Comment (cballevre) [inline react/Buttons/Readme.md:1]: Is it deliberate to remove this line?

**Review by cballevre — APPROVED** [review]
Thank you for documenting this point 👍

File: react/Buttons/Readme.md (hunk 1 above)
Comment (Ldoppea) [reply react/Buttons/Readme.md:1]: Yes, I explained why on [the first commit](https://github.com/cozy/cozy-ui/pull/2593/commits/b38b6a02259aa87143677da0f4318db1bc9a6ff0) 😋

**Review by JF-Cozy — APPROVED** [review]
pour les commit, ce ne sont pas des "chore" mais des "docs" :)

**Pull request merged**